util/oidc/oidc.go

```diff
@@ -640,6 +642,37 @@ func createClaimsAuthenticationRequestParameter(requestedClaims map[string]*oidc
 	return oauth2.SetAuthURLParam("claims", string(claimsRequestRAW)), nil
 }
 
+// AddGroupsFromUserInfo takes a claims object and adds groups claim from userinfo endpoint if available
```

Comment:
What should happen when this function isn't able to reach the UserInfoEndpoint?

In the past it used to invalidate the session and let the user login (because the most likely case for not being able to access the UserInfoEndpoint was a missing accessToken from the user).

However the WithAuthMiddleware has no real error handling possibilities, that's why I changed it to silently return the incoming claims and not do anything. Thankfully that let's a test for server.go to fail since it's expecting the current behavior...

Reply:
I added error handling to fail fast, invalidating the session if the feature is enabled and we aren't able to fetch the claims.